src/python/pants/engine/environment.py

```diff
@@ -6,8 +6,12 @@
 
 from pants.engine.engine_aware import EngineAwareParameter
 
+# Reserved sentinel value directing Pants to find applicable local environment.
 LOCAL_ENVIRONMENT_MATCHER = "__local__"
 
+# Reserved sentinel value representing execution within the workspace and not local sandbox.
+LOCAL_WORKSPACE_ENV_NAME = "__local_workspace__"
```

Comment:
This sentinel value was added so tests in this PR could make use of the workspace process execution strategy even without the next PR in the series (which adds the `workspace_environment` type). Maybe it should be renamed so this support is not in the public API?

Reply:
cc @benjyw

Reply:
I prefixed the symbol with underscore to make it clear that the symbol is not public API, and added a comment to that effect.